### DISCUSSION TITLE
Downgrade Sonoff Error

### DISCUSSION THREAD

**Discussion by ricardobortolotti** [arendst/Tasmota#20546] [Support request] [answered]
Hello!

I have a old version of sonoff, and it was working very well with Tasmota 12.1.1 version.

I did the OTA upgrade for 13.0.3 version and the sonoff stop working. Looking in my router, the sonoff connect and disconect instantily all the time.

So, I'm trying to do the downgrade version to 12.1.1 again, but I always recived the mensage: File is too large.
I'm using the minimal version with bin.gz

My sonoff is behind of a wall, and I don't phisical access to flash the sonoff again.

Can anyone help me, please?

Thak you!

**Comment by Jason2866** [comment]
Downgrade is and was never supported. Only way to recover is to flash via serial. Before flashing an old version a full flash erase is needed.

**Comment by ricardobortolotti** [comment]
Hello!
Thank you Jason for your support!
I didn’t know about downgrade rules!

It’s normal my old version of sonoff don’t work with the lasted Tasmota version?

I really need a solution, because I can’t access the equipment behind the wall.

Has any thing that I can do for use my Sonoff with this Tastoma version?

Thank you a lot!

**Reply by Jason2866** [reply]
Sonoff devices do often have a weak power supply. Since in every power supply Caps are used, the power supply gets bader over time. Caps loose capacity over time.
Newer Tasmota version do need a bit more current than older versions. This little bit can be too much.
It is no good idea to place electronic devices unreachable. The will get defect. It is just a question of time when this will happen. In my experience Sonoff devices die after a few years.
Latest Tasmota does run fine with fabric new devices. Flashing something new on old devices is always a bit of risk, since the health state of the power supply is unknown.

To say it clear, if you can't reach the device you can't do nothing at all.

If the device is still reachable and you can initiate an upgrade you may have luck by using
https://github.com/tasmota/ESP8266MinimalHTTPUpdater

**Reply by sfromis** [reply]
Reading your words, it seems like you have access some of the time, and then (if you get a chance), you could OTA "upgrade" to https://github.com/tasmota/ESP8266MinimalHTTPUpdater/ and after that do an OTA upgrade to some older Tasmota version. Tasmota also uses a bit less power using the lite variant of binaries, but I'd suggest planning for replacing the degraded device(s).

**Reply by ricardobortolotti** [reply]
Thanks for your answers.

I just have access in initial config, where I can set the wi-fi network.
![image](https://github.com/arendst/Tasmota/assets/2806798/a83603f9-47a1-4c3d-acef-d971ef6f56ed)


In this part, if I set the network, the equipment start the problem with connection.

I can upload file for restore, but I don’t have any menu for OTA URLs.

So, I think that is not possible to do by OTA method.

**Reply by barbudor** [reply, marked as answer]
You need to append `/up` to the url as in `http://192.168.4.1/up` to access the upgrade menu
While not connected to your WLAN network, to flash Jason's upgrader, you need first to download the bin.gz from [here](https://github.com/tasmota/ESP8266MinimalHTTPUpdater/blob/master/build_output/firmware/updater.bin.gz) using hte download button on the right
![image](https://github.com/arendst/Tasmota/assets/2996042/71df95f4-6a14-42f1-aca9-dce5fc6d59e0)

Then connect to `http://192.168.4.1/up` and use the 2nd part: "Upgrade by file upload" to install Jason's flasher

From there you should be able to reflash any Tasmota version

**Reply by Jason2866** [reply]
But do NOT flash the minimal variant! Best is to try the lite version.

**Reply by ricardobortolotti** [reply]
@barbudor I did the upload by the URL /up
After the Upload, I found a new WiFi network "UpdateMe", but I can't connect.

I had tryied in my macbook and iPhone, and don't work in both. They try to connect, but not complete.

I will try in another device, but I think that not work how was expected...

I'm thinking that this Sonoff is broken.

Thank you for your help.

**Reply by barbudor** [reply]
Have you read the instructions of Jason's uploader? 
Including the sentence about the AP wifi password being 'recovery'?

**Reply by ricardobortolotti** [reply]
Yes, I had trying this password and works, but the connection was aborted.
I didn’t do the instruction about console part: When running Tasmota: Do command SetOption78 1in Console before OTA.

Can be it the problem?

**Reply by Jason2866** [reply]
No, since the minimal updater is running and has opened the AP

**Reply by Jason2866** [reply]
Tried with which device to connect?
Only 2.4 GHz. Devices with 2.4 and 5 GHz maybe problematic

**Reply by sfromis** [reply]
The part about using `SetOption78 1` is not relevant to anything near current Tasmota versions, by now only very old ones.

**Reply by ricardobortolotti** [reply]
It worked!!
I had access with an old computer with only 2.4 Ghz Wi-Fi network.
After that I sent a lite version of Tasmota.
Thanks a lot @barbudor and @Jason2866, you are amazing. ❤️
